src/Drupal/Bootstrap.php

```diff
@@ -223,4 +224,17 @@ protected function loadExtension(Extension $extension): void
             // This can happen when drupal_get_path or drupal_get_filename are used outside of the scope of a function.
         }
     }
+
+    protected function loadAndCatchErrors(string $path): void
+    {
+        try {
+            require $path;
+        } catch (ContainerNotInitializedException $e) {
+            // This can happen when drupal_get_path or drupal_get_filename are used outside of the scope of a function.
+            @trigger_error("$path invoked the Drupal container outside of the scope of a function or class method. It was skipped for analysis.", E_USER_WARNING);
```

Comment:
proof of concept catching these to be replayed in drupal-check or another runner.